core/trino-main/src/main/java/io/trino/sql/routine/SqlRoutineCompiler.java

```diff
@@ -287,8 +287,8 @@ private class BytecodeVisitor
         private final Map<LambdaDefinitionExpression, CompiledLambda> compiledLambdaMap;
         private final Map<IrVariable, Variable> variables;
 
-        private final Map<IrLabel, LabelNode> continueLabels = new IdentityHashMap<>();
-        private final Map<IrLabel, LabelNode> breakLabels = new IdentityHashMap<>();
+        private final Map<IrLabel, LabelNode> continueLabels = new HashMap<>();
+        private final Map<IrLabel, LabelNode> breakLabels = new HashMap<>();
```

Comment:
this won't work when function body has same label multiple times.
see `TestSqlFunctions.testReuseLabels`